mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/ReconnectionStrategyTypeBuilder.java

```diff
@@ -29,16 +28,38 @@
   public static final String RECONNECT_FOREVER_ALIAS = "reconnect-forever";
   public static final String BLOCKING = "blocking";
   public static final String RECONNECTION_STRATEGY = "ReconnectionStrategy";
+  public static final String RECONNECTION_CONFIG = "Reconnection";
+
+  /**
+   * @return a {@link MetadataType} representation of a reconnection configuration
+   */
+  public MetadataType buildReconnectionConfigType() {
+    BaseTypeBuilder typeBuilder = create(JAVA);
+    ObjectTypeBuilder type = create(JAVA).objectType()
+        .id(Object.class.getName())
+        .with(new TypeAliasAnnotation(RECONNECTION_CONFIG))
+        .with(new InfrastructureTypeAnnotation());
+
+    addBooleanField(type, typeBuilder, "failsDeployment",
+                    "When the application is deployed, a connectivity test is performed on all connectors. If set to " +
```

Comment:
You defined this in a constant (RECONNECTION_CONFIG_PARAMETER_DESCRIPTION) already.